controllers/storagecluster/prometheus.go

```diff
@@ -79,10 +79,11 @@ func getPrometheusRuleSpecFrom(filePath string) (*monitoringv1.PrometheusRuleSpe
 	if err != nil {
 		return nil, fmt.Errorf("'%s' not readable", filePath)
 	}
-	ruleSpec := monitoringv1.PrometheusRuleSpec{}
-	if err := k8sYAML.NewYAMLOrJSONDecoder(bytes.NewBufferString(string(fileContent)), 1000).Decode(&ruleSpec); err != nil {
```

Comment:
@jarrpa Apologies for the late reply on this, I was out sick. The above code is expected to decode the rule file into an object and also get the spec out of it and put it in the `ruleSpec` variable. It fails to do so as the object that gets created is of type struct `monitoringv1.PrometheusRule{} `and thus the variable ruleSpec remains empty as it of type `monitoringv1.PrometheusRuleSpec{}`

I am trying to do is, to first convert the rule file into its matching type object i.e `monitoringv1.PrometheusRule{`} and then take the spec out to return it.

Reply:
Sure. Include something about this in the commit message. :slightly_smiling_face:

Reply:
@jarrpa Thank You. I added the short version of it to commit and updated the PR description with the same. Please review.